crates/ruff/src/rules/refurb/rules/repeated_append.rs

```diff
@@ -0,0 +1,377 @@
+    let appends = &group.appends;
+
+    assert!(!appends.is_empty());
+    let first = appends.first().unwrap();
+
+    assert!(appends
+        .iter()
+        .all(|append| append.binding.source == first.binding.source));
+
+    // Here we construct `var.extend((elt1, elt2, ..., eltN))
+    //
+    // Each eltK comes from an individual `var.append(eltK)`.
+    let elts: Vec<ast::Expr> = appends
+        .iter()
+        .map(|append| append.argument.clone())
+        .collect();
+    // Join all elements into a tuple: `(elt1, elt2, ..., eltN)`
+    let tuple = ast::ExprTuple {
+        elts,
+        ctx: ast::ExprContext::Load,
+        range: TextRange::default(),
+    };
+    // Make `var.extend`.
+    // NOTE: receiver is the same for all appends and that's why we can take the first.
+    let attr = ast::ExprAttribute {
+        value: Box::new(first.receiver.clone().into()),
+        attr: ast::Identifier::new("extend".to_string(), TextRange::default()),
+        ctx: ast::ExprContext::Load,
+        range: TextRange::default(),
+    };
+    // Make the actual call `var.extend((elt1, elt2, ..., eltN))`
+    let call = ast::ExprCall {
+        func: Box::new(attr.into()),
+        arguments: ast::Arguments {
+            args: vec![tuple.into()],
+            keywords: vec![],
+            range: TextRange::default(),
+        },
+        range: TextRange::default(),
+    };
+    // And finally, turn it into a statement.
+    let stmt = ast::StmtExpr {
+        value: Box::new(call.into()),
+        range: TextRange::default(),
+    };
+    generator.stmt(&stmt.into())
+}
+
+fn match_append<'a>(checker: &'a Checker, stmt: &'a Stmt) -> Option<Append<'a>> {
+    let Stmt::Expr(ast::StmtExpr { value, .. }) = stmt else {
+        return None;
+    };
+
+    let Expr::Call(ast::ExprCall {
+        func, arguments, ..
+    }) = value.as_ref()
+    else {
+        return None;
+    };
+
+    if arguments.args.len() != 1 {
+        return None;
+    }
+
+    let Expr::Attribute(ast::ExprAttribute { value, attr, .. }) = func.as_ref() else {
+        return None;
+    };
+
+    if attr != "append" {
+        return None;
+    }
+
+    let Expr::Name(receiver @ ast::ExprName { id: name, .. }) = value.as_ref() else {
+        return None;
+    };
+
+    let scope = checker.semantic().current_scope();
+    let bindings: Vec<&'a Binding> = scope
+        .get_all(name)
+        .map(|binding_id| checker.semantic().binding(binding_id))
+        .collect();
+
+    // Maybe it is too strict of a limitation, but it seems reasonable.
+    if bindings.len() != 1 {
+        return None;
+    }
+
+    let binding = bindings[0];
+    if binding.source.is_none() || !is_list(checker, binding, name) {
+        return None;
+    }
+
+    Some(Append {
+        receiver,
+        binding,
+        stmt,
+        argument: arguments.args.first().unwrap(),
+    })
+}
+
+fn is_list<'a>(checker: &'a Checker, binding: &'a Binding, name: &str) -> bool {
```

Comment:
This got folded into a more general trait in a future PR, right?

Reply:
Yep, the next PR folds it and adds is_dict, and the one after that is_set